src/libraries/Microsoft.PowerFx.Core/Utils/StackMarker.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    internal struct StackMarker
+    {
+        private readonly int _depth;
+        private readonly int _maxCallDepth;
```

Comment:
nit - this could actually be a single counter rather than 2. (could be relevant since we're passing as parameters on a stack). 

Start at _maxCallDepth and count towards 0. Fail if < 0 .

Reply:
That might make it harder to debug right?

Reply:
That is **smart** why did I not think of that.